deeppavlov/core/trainers/nn_trainer.py

```diff
@@ -146,13 +168,41 @@ def _improved(op):
         self.losses = []
         self.start_time: Optional[float] = None
 
-        if self.tensorboard_log_dir is not None:
-            self.tb_train_writer = self._tf.summary.FileWriter(str(self.tensorboard_log_dir / 'train_log'))
-            self.tb_valid_writer = self._tf.summary.FileWriter(str(self.tensorboard_log_dir / 'valid_log'))
+        if self.stdlogger_idx is not None:
+            self.std_logger = StdLogger(stdlogging=True)
+
+        if self.tensorboard_idx is not None:
+            self.tensorboard_logger = TensorboardLogger(
+                log_dir=self.logger[self.tensorboard_idx]["log_dir"]
+            )
+
+        if self.wandblogger_idx is not None:
+            if WandbLogger.login(
+                API_Key=self.logger[self.wandblogger_idx].get("API_Key", None),
+                relogin=True,
+            ):
+                if self.log_every_n_epochs > 0 or self.val_every_n_epochs > 0:
+                    self.wandb_logger = WandbLogger(
+                        log_on="epochs",
+                        commit_on_valid=self.val_every_n_epochs > 0,
+                        **self.logger[self.wandblogger_idx].get("init", None),
+                    )
+                    if self.wandb_logger.init_succeed == False:
+                        self.wandblogger_idx = None
+                elif self.log_every_n_batches > 0 or self.val_every_n_batches > 0:
+                    self.wandb_logger = WandbLogger(
+                        log_on="batches",
+                        commit_on_valid=self.val_every_n_batches > 0,
+                        **self.logger[self.wandblogger_idx].get("init", None),
+                    )
+                    if self.wandb_logger.init_succeed == False:
+                        self.wandblogger_idx = None
+            else:
+                self.wandblogger_idx = None
```

Comment:
remove those lines from nn_trainer.py and move to __init__ methods of according classes.